### YamlMime:ManagedReference
items:
- uid: System.Workflow.Runtime.WorkflowTerminatedEventArgs
  id: WorkflowTerminatedEventArgs
  children:
  - System.Workflow.Runtime.WorkflowTerminatedEventArgs.Exception
  langs:
  - csharp
  name: WorkflowTerminatedEventArgs
  nameWithType: WorkflowTerminatedEventArgs
  fullName: System.Workflow.Runtime.WorkflowTerminatedEventArgs
  type: Class
  summary: "Provides data for the <xref href=&quot;System.Workflow.Runtime.WorkflowRuntime.WorkflowTerminated&quot;></xref> event."
  remarks: "> [!NOTE] >  [!INCLUDE[DeprecatedContent](~/add/includes/ajax-current-ext-md.md)]       A workflow instance may be terminated in one of three ways: the host may call <xref:System.Workflow.Runtime.WorkflowInstance.Terminate%2A?displayProperty=fullName>, a <xref:System.Workflow.ComponentModel.TerminateActivity> activity may be invoked from inside the workflow instance, or the workflow runtime engine may terminate the instance because of an unhandled exception. When the workflow instance is terminated, the workflow runtime engine raises the <xref:System.Workflow.Runtime.WorkflowRuntime.WorkflowTerminated?displayProperty=fullName> event. The host may subscribe to this event by adding an event handler to <xref:System.Workflow.Runtime.WorkflowRuntime.WorkflowTerminated?displayProperty=fullName>."
  example:
  - "The following code example demonstrates an event handler method that takes WorkflowTerminatedEventArgs as a parameter. The `OnWorkflowTerminated` method is called when the <xref:System.Workflow.Runtime.WorkflowRuntime.WorkflowTerminated> event is raised. The code uses the <xref:System.Workflow.Runtime.WorkflowTerminatedEventArgs.Exception%2A> property get the message associated with the exception that caused the workflow to terminate.  \n  \n This code example is part of the Sequential Workflow with Parameters SDK Sample from the Program.cs file. For more information, see [Sequential Workflow with Parameters](http://msdn.microsoft.com/en-us/17157f2d-8fed-4848-a386-e4efa381b66b).  \n  \n [!code-cs[WF_Samples#193](~/add/codesnippet/csharp/wf_snippets/snippets19.cs#193)]\n [!code-vb[WF_Samples#193](~/add/codesnippet/visualbasic/wf_snippets/snippets19.vb#193)]"
  syntax:
    content: >-
      [System.Obsolete("The System.Workflow.* types are deprecated.  Instead, please use the new types from System.Activities.*")]

      public class WorkflowTerminatedEventArgs : System.Workflow.Runtime.WorkflowEventArgs
  inheritance:
  - System.EventArgs
  - System.Workflow.Runtime.WorkflowEventArgs
  implements: []
  inheritedMembers:
  - System.Workflow.Runtime.WorkflowEventArgs.WorkflowInstance
  platform:
  - net462
- uid: System.Workflow.Runtime.WorkflowTerminatedEventArgs.Exception
  id: Exception
  parent: System.Workflow.Runtime.WorkflowTerminatedEventArgs
  langs:
  - csharp
  name: Exception
  nameWithType: WorkflowTerminatedEventArgs.Exception
  fullName: System.Workflow.Runtime.WorkflowTerminatedEventArgs.Exception
  type: Property
  assemblies:
  - System.Workflow.Runtime
  namespace: System.Workflow.Runtime
  summary: "Gets the exception that caused the workflow instance to be terminated."
  remarks: "When the workflow instance is terminated because of an unhandled exception, Exception contains the unhandled exception.       When the workflow instance is terminated by either a host call to <xref:System.Workflow.Runtime.WorkflowInstance.Terminate%2A?displayProperty=fullName> or by a <xref:System.Workflow.ComponentModel.TerminateActivity> activity,    Exception contains a <xref:System.Workflow.ComponentModel.WorkflowTerminatedException> that has its <xref:System.Exception.Message%2A> property set to a description of the reason for the termination. If the host terminates the workflow instance, it supplies this description in the `string` parameter to <xref:System.Workflow.Runtime.WorkflowInstance.Terminate%2A?displayProperty=fullName>. If the workflow instance is terminated by a <xref:System.Workflow.ComponentModel.TerminateActivity>, the description is supplied by <xref:System.Workflow.ComponentModel.TerminateActivity.Error%2A?displayProperty=fullName>.      > [!NOTE] >  Although in the case of a <xref:System.Workflow.Runtime.WorkflowRuntime.WorkflowTerminated?displayProperty=fullName> event that is not due to an unhandled exception, the workflow runtime engine encapsulates a description of the reason for the termination in the <xref:System.Exception.Message%2A> property of a <xref:System.Workflow.ComponentModel.WorkflowTerminatedException>, it does not necessarily throw this exception."
  example:
  - "The following code example demonstrates how you can use the Exception property to obtain information about an exception that caused a workflow to terminate. The `OnWorkflowTerminated` method takes a <xref:System.Workflow.Runtime.WorkflowTerminatedEventArgs> as a parameter. This method is called when the <xref:System.Workflow.Runtime.WorkflowRuntime.WorkflowTerminated> event is raised. The code uses the Exception property get the message associated with the exception that caused the workflow to terminate.  \n  \n This code example is part of the Sequential Workflow with Parameters SDK Sample from the Program.cs file. For more information, see [Sequential Workflow with Parameters](http://msdn.microsoft.com/en-us/17157f2d-8fed-4848-a386-e4efa381b66b).  \n  \n [!code-cs[WF_Samples#193](~/add/codesnippet/csharp/wf_snippets/snippets19.cs#193)]\n [!code-vb[WF_Samples#193](~/add/codesnippet/visualbasic/wf_snippets/snippets19.vb#193)]"
  syntax:
    content: public Exception Exception { get; }
    return:
      type: System.Exception
      description: "The <xref:System.Exception> that caused the workflow instance to be terminated."
  overload: System.Workflow.Runtime.WorkflowTerminatedEventArgs.Exception*
  exceptions: []
  platform:
  - net462
references:
- uid: System.Workflow.Runtime.WorkflowEventArgs
  isExternal: false
  name: System.Workflow.Runtime.WorkflowEventArgs
- uid: System.Workflow.Runtime.WorkflowTerminatedEventArgs.Exception
  parent: System.Workflow.Runtime.WorkflowTerminatedEventArgs
  isExternal: false
  name: Exception
  nameWithType: WorkflowTerminatedEventArgs.Exception
  fullName: System.Workflow.Runtime.WorkflowTerminatedEventArgs.Exception
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Workflow.Runtime.WorkflowTerminatedEventArgs.Exception*
  parent: System.Workflow.Runtime.WorkflowTerminatedEventArgs
  isExternal: false
  name: Exception
  nameWithType: WorkflowTerminatedEventArgs.Exception
